Framework/Algorithms/src/SumOverlappingTubes.cpp

```diff
@@ -212,21 +214,17 @@ void SumOverlappingTubes::getHeightAxis(const std::string &componentName) {
       (m_outputType != "1D" && heightBinning.size() == 2)) {
     // Try to get the component. It should be a tube with pixels in the
     // y-direction, the height bins are then taken as the detector positions.
-    const auto &ws = m_workspaceList.front();
-    const auto &inst = ws->getInstrument()->baseInstrument();
-    const auto comp = inst->getComponentByName(componentName);
-    if (!comp)
-      throw std::runtime_error("Component " + componentName +
-                               " could not be found.");
-    const auto &compAss = dynamic_cast<const ICompAssembly &>(*comp);
-    std::vector<IComponent_const_sptr> children;
-    compAss.getChildren(children, false);
-    for (const auto &child : children) {
-      const auto posY = child->getPos().Y();
+    const auto &componentInfo = m_workspaceList.front()->componentInfo();
+    const auto &detectorInfo = m_workspaceList.front()->detectorInfo();
+    const auto componentIndex = componentInfo.indexOf(componentName);
+    const auto &detsInSubtree =
+        componentInfo.detectorsInSubtree(componentIndex);
+    for (const auto detIndex : detsInSubtree) {
+      const auto posY = detectorInfo.position({detIndex, 0}).Y();
```

Comment:
You can make this call directly through `componentInfo`
```cpp
const auto posY = componentInfo.position({detIndex, 0}).Y()
```

Reply:
Is this possible because `detectorIndex == componentIndex`?

Reply:
Yes. A detector index is guaranteed to be the same a component index http://docs.mantidproject.org/nightly/concepts/InstrumentAccessLayers.html#access-to-components-and-working-with-detectors